src/fungible_token.rs

```diff
@@ -0,0 +1,368 @@
+        }
+    }
+
+    pub fn internal_storage_balance_of(&self, account_id: AccountId) -> Option<StorageBalance> {
+        if self.accounts_contains_key(account_id) {
+            Some(StorageBalance {
+                total: self.storage_balance_bounds().min,
+                available: 0,
+            })
+        } else {
+            None
+        }
+    }
+
+    pub fn storage_balance_of(&self, account_id: AccountId) -> Option<StorageBalance> {
+        self.internal_storage_balance_of(account_id)
+    }
+
+    // `registration_only` doesn't affect the implementation for vanilla fungible token.
+    #[allow(unused_variables)]
+    pub fn storage_deposit(
+        &mut self,
+        account_id: Option<AccountId>,
+        registration_only: Option<bool>,
+    ) -> StorageBalance {
+        let amount: Balance = sdk::attached_deposit();
+        let predecessor_account_id = String::from_utf8(sdk::predecessor_account_id()).unwrap();
+        let account_id = account_id.unwrap_or(predecessor_account_id);
+        if self.accounts_contains_key(account_id.clone()) {
+            #[cfg(feature = "log")]
+            sdk::log("The account is already registered, refunding the deposit".into());
+            if amount > 0 {
+                let promise0 = sdk::promise_batch_create(&sdk::predecessor_account_id());
+                sdk::promise_batch_action_transfer(promise0, amount);
+            }
+        } else {
+            let min_balance = self.storage_balance_bounds().min;
+            if amount < min_balance {
+                #[cfg(feature = "log")]
+                sdk::panic_utf8(
+                    "The attached deposit is less than the minimum storage balance".as_bytes(),
+                );
+            }
+
+            self.internal_register_account(account_id.clone());
+            let refund = amount - min_balance;
+            if refund > 0 {
+                let promise0 = sdk::promise_batch_create(&sdk::predecessor_account_id());
+                sdk::promise_batch_action_transfer(promise0, refund);
+            }
+        }
+        self.internal_storage_balance_of(account_id).unwrap()
+    }
+
+    pub fn storage_unregister(&mut self, force: Option<bool>) -> bool {
+        self.internal_storage_unregister(force).is_some()
+    }
+
+    pub fn storage_withdraw(&mut self, amount: Option<u128>) -> StorageBalance {
+        sdk::assert_one_yocto();
+        let predecessor_account_id = String::from_utf8(sdk::predecessor_account_id()).unwrap();
+        if let Some(storage_balance) = self.internal_storage_balance_of(predecessor_account_id) {
+            match amount {
+                Some(amount) if amount > 0 => {
+                    sdk::panic_utf8(b"ERR_WRONG_AMOUNT");
+                }
+                _ => storage_balance,
+            }
+        } else {
+            sdk::panic_utf8(b"ERR_ACCOUNT_NOT_REGISTERED");
+        }
+    }
+
+    fn ft_key(&self, account_id: AccountId) -> Vec<u8> {
```

Comment:
Performance nit: I think most of the calls to `clone` could be removed if this function (and essentially all the other functions in this module) took `&str` instead of `AccountId`.

Reply:
I think you mean reference instead of move value. It's a really small overhead. But I have got a lot of issues when I tried `&str`. And I tried it. But return String. So if it'll a real point it will possible to spend time on that. But IMHO for that concrete point it'll really small overhead.

Reply:
I agree, it is likely a small overhead. I don't think it should block the PR.

This is a change I would be happy to contribute though. I'm sure you have more important things to be doing with regards to this work, but I could help in this small way if you think that's ok.

Reply:
Awesome! Really thanks! Yes, please.

Reply:
Bit just keep in mind, that for now better place for changes is #24

Reply:
I have created https://github.com/aurora-is-near/aurora-engine/pull/29 which targets the branch #24 is working from.